### PR TITLE
CI(build-tools): don't install Postgres 14

### PR DESCRIPTION
**Pull request by bayandin**
## Problem

We install Postgres 14 in `build-tools` image, but we don't really need it. We use Postgres binaries that we build ourselves. 

## Summary of changes
- Remove Postgresql 14 installation from `build-tools` image

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics? if so did you add the relevant metrics to the dashboard?
- [ ] If this PR requires public announcement, mark it with /release-notes label and add several sentences in this section.

## Checklist before merging

- [ ] Do not forget to reformat commit message to not include the above checklist

### CODE REVIEW

**Review by Bodobolero — APPROVED** [review]
LGTM, but we don't know if all users of that container that need Postgres-client add their own built or downloaded postgres client binaries to the path but expect the prior behavior that it is already there. So after the merging the PR we need to watch the different CI runs for potential failures caused by this - it is easily detectable by our assertion that the postgres client tool path was resolved correctly, though

**Comment by alexanderlaw** [comment]
I've discovered that test_proxy.py::test_link_auth and test_proxy_metric_collection.py::test_proxy_metric_collection fail when psql's directory is not in PATH (postgres client utilities not installed).

I run the test with:
`POETRY_VIRTUALENVS_PATH=virtualenvs DEFAULT_PG_VERSION=17 BUILD_TYPE=debug poetry run pytest test_runner/regress/test_proxy.py`
and get:
```
test_runner/regress/test_proxy.py::test_link_auth[debug-pg17]

__________________________ test_link_auth[debug-pg17] __________________________

vanilla_pg = <fixtures.neon_fixtures.VanillaPostgres object at 0x70ffb0c8dc40>
link_proxy = <fixtures.neon_fixtures.NeonProxy object at 0x70ffb0061a00>

    @pytest.mark.asyncio 
    async def test_link_auth(vanilla_pg: VanillaPostgres, link_proxy: NeonProxy):
        """
        Check the Link auth flow: a lightweight auth method which delegates
        all necessary checks to the console by sending client an auth URL.
        """
        
>       psql = await PSQL(host=link_proxy.host, port=link_proxy.proxy_port).run("select 42")

...
        search_path = None
        if (d := os.getenv("POSTGRES_DISTRIB_DIR")) is not None and (
            v := os.getenv("DEFAULT_PG_VERSION")
        ) is not None:
            search_path = Path(d) / f"v{v}" / "bin"

        full_path = shutil.which(path, path=search_path)
>       assert full_path is not None
E       assert None is not None

test_runner/fixtures/neon_fixtures.py:3067: AssertionError
```
(The POSTGRES_DISTRIB_DIR environmental variable is empty.)

That is, these tests still use psql from OS, not built one.

A dirty test fix like this (with the corresponding imports added):
```
 @pytest.mark.asyncio
-async def test_link_auth(vanilla_pg: VanillaPostgres, link_proxy: NeonProxy):
+async def test_link_auth(vanilla_pg: VanillaPostgres, link_proxy: NeonProxy, pg_distrib_dir: Path):
     """
     Check the Link auth flow: a lightweight auth method which delegates
     all necessary checks to the console by sending client an auth URL.
     """
 
+    os.environ["POSTGRES_DISTRIB_DIR"] = str(pg_distrib_dir)
```
makes the test_proxy.py test pass for me.

**Pull request merged**
